src/networkcloud/azext_networkcloud/aaz/latest/networkcloud/baremetalmachine/_power_off.py

```diff
@@ -0,0 +1,326 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#
+# Code generated by aaz-dev-tools
+# --------------------------------------------------------------------------------------------
+
+# pylint: skip-file
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "networkcloud baremetalmachine power-off",
+    is_experimental=True,
+)
+class PowerOff(AAZCommand):
+    """Power off the provided bare metal machine.
+
+    :example: Power off bare metal machine
+        az networkcloud baremetalmachine power-off --bare-metal-machine-name "bareMetalMachineName" --skip-shutdown "True" --resource-group "resourceGroupName"
+    """
+
+    _aaz_info = {
+        "version": "2022-12-12-preview",
+        "resources": [
+            ["mgmt-plane", "/subscriptions/{}/resourcegroups/{}/providers/microsoft.networkcloud/baremetalmachines/{}/poweroff", "2022-12-12-preview"],
+        ]
+    }
+
+    AZ_SUPPORT_NO_WAIT = True
+
+    def _handler(self, command_args):
+        super()._handler(command_args)
+        return self.build_lro_poller(self._execute_operations, self._output)
+
+    _args_schema = None
+
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        if cls._args_schema is not None:
+            return cls._args_schema
+        cls._args_schema = super()._build_arguments_schema(*args, **kwargs)
+
+        # define Arg Group ""
+
+        _args_schema = cls._args_schema
+        _args_schema.bare_metal_machine_name = AAZStrArg(
+            options=["-n", "--name", "--bare-metal-machine-name"],
+            help="The name of the bare metal machine.",
+            required=True,
+            id_part="name",
+            fmt=AAZStrArgFormat(
+                pattern="^([a-zA-Z0-9][a-zA-Z0-9]{0,62}[a-zA-Z0-9])$",
+            ),
+        )
+        _args_schema.resource_group = AAZResourceGroupNameArg(
+            required=True,
+        )
+
+        # define Arg Group "BareMetalMachinePowerOffParameters"
+
+        _args_schema = cls._args_schema
+        _args_schema.skip_shutdown = AAZStrArg(
```

Comment:
Please verify this property is a string type in API.

Reply:
This is the link to the API specification for "skip_shutdown"  https://github.com/Azure/azure-rest-api-specs/blame/main/specification/networkcloud/resource-manager/Microsoft.NetworkCloud/preview/2022-12-12-preview/networkcloud.json#L6653 which is of type "string". 

Please let us know if we are missing something.

Reply:
Thanks for your clarification.